src/main/java/picard/sam/BamErrorMetric/BaseErrorCalculation.java

```diff
@@ -0,0 +1,215 @@
+        long nBothDisagreeWithReference = 0;
+        long nDisagreeWithRefAndMate = 0;
+        long nThreeWaysDisagreement = 0;
+        long nTotalBasesWithOverlappingReads = 0;
+
+        @Override
+        public void addBase(final SamLocusIterator.RecordAndOffset recordAndOffset, final SAMLocusAndReferenceIterator.SAMLocusAndReference locusAndRef) {
+            super.addBase(recordAndOffset, locusAndRef);
+            final byte readBase = recordAndOffset.getReadBase();
+            final SAMRecord record = recordAndOffset.getRecord();
+            final SamLocusIterator.RecordAndOffset mate = locusAndRef.getRecordAndOffsets()
+                    .stream()
+                    .filter(putative -> areReadsMates(record, putative.getRecord()))
+                    .findFirst()
+                    .orElse(null);
+
+            // we are only interested in bases for which the mate read also has a base over the same locus
+            if (mate == null) return;
+            // both bases need to be called for this error calculation
+
+            final byte mateBase = mate.getReadBase();
+            if (SequenceUtil.isNoCall(readBase)) return;
+            if (SequenceUtil.isNoCall(mateBase)) return;
+
+            nTotalBasesWithOverlappingReads++;
+
+            // Only bases that disagree with the reference are counted as errors.
+            if (SequenceUtil.basesEqual(readBase, locusAndRef.referenceBase)) return;
+
+            final boolean agreesWithMate = SequenceUtil.basesEqual(readBase, mateBase);
+            final boolean mateAgreesWithRef = SequenceUtil.basesEqual(mateBase, locusAndRef.referenceBase);
+
+            if (agreesWithMate) {
+                nBothDisagreeWithReference++;
+            } else if (mateAgreesWithRef) {
+                nDisagreeWithRefAndMate++;
+            } else {
+                nThreeWaysDisagreement++;
+            }
+        }
+
+        @Override
+        public String getSuffix() {
+            return "overlapping_error";
+        }
+
+        @Override
+        public ErrorMetrics.OverlappingErrorMetric getMetric() {
+            return new ErrorMetrics.OverlappingErrorMetric("", totalBases, nTotalBasesWithOverlappingReads, nDisagreeWithRefAndMate,
+                    nBothDisagreeWithReference, nThreeWaysDisagreement);
+        }
+
+        private boolean areReadsMates(final SAMRecord read1, final SAMRecord read2) {
+            // must have same name
+            if (!read1.getReadName().equals(read2.getReadName())) return false;
+            // must be paired
+            if (!read1.getReadPairedFlag()) return false;
+            // one must be first while the other is not
+            if (!read1.getFirstOfPairFlag() ^ read2.getFirstOfPairFlag()) return false;
+            // one must be second while the other is not
+            if (!read1.getSecondOfPairFlag() ^ read2.getSecondOfPairFlag()) return false;
```

Comment:
how about:
```
if ((read1.getFirstOfPairFlag() && read2. getSecondOfPairFlag()) || (read1.getSecondOfPairFlag() && read2. getFirstOfPairFlag())
```
You're not wrong, it just might link the fact that one should be r1 and the other r2.